utilities/backup/backup_engine_test.cc

```diff
@@ -4790,6 +4790,77 @@ TEST_F(BackupEngineTest, IOBufferSize) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
 }
 
+// Test stopping backup at different points in the backup lifecycle
+TEST_F(BackupEngineTest, StopBackupAtDifferentStages) {
+  const int keys_iteration = 5000;
+
+  // Test different threshold values:
+  // 0 = stop immediately (before work starts)
+  // 3, 10, 50 = stop after some work (backup may complete successfully)
+  std::vector<int> stop_thresholds = {0, 3, 10, 50};
+
+  for (int stop_after_calls : stop_thresholds) {
+    OpenDBAndBackupEngine(true);
+    FillDB(db_.get(), 0, keys_iteration);
+
+    // Set up sync point to stop backup after a certain number of calls
+    int call_count = 0;
```

Comment:
I see this only works reliably in single-threaded cases. We should probably enable multi-threaded backup here (`max_background_operations`) to expand the logic covered, and make this a RelaxedAtomic.